contracts/gateway/INTEGRATION.md

```diff
@@ -229,19 +231,44 @@ event SignersRotated(uint256 indexed epoch, bytes32 indexed signersHash, bytes s
+5. **Storage limit**: The gateway contract stores message execution status. The number of messages stored is arbitrary and will grow over time. Messages that have been executed can't be deleted to prevent replay of approvals. Implementations should support a practically unlimited number of messages to be stored (e.g. `2^64`).
+6. **Event Retention**: Cross-chain messages and signer rotations are verified by fetching events from the chain's node. We recommend that events can be retained for 2 months, and at a minimum for 2 weeks to allow processing of stuck messages, or signer rotations. These could be stuck due to being missed by the relayer, or due to insufficient gas paid by the app (but adds the gas later to process it).
+
+The limit recommendations are summarized in the following table. The limits applicable for the EVM gateway on Ethereum (assuming a block gas limit of 30M) are also shown for comparison.
+
+| Limit | Minimum | Recommended | Ethereum Gateway |
+|-|-|-|-|
+| Cross-chain Message Size | 16 KB | > 64 KB | > 1 MB |
+| Chain Name Length | < 20 ASCII chars | - | - |
+| Signer Set Size | 40 signers | 100 signers | > 200 |
+| Signature Verification | 27 signatures | 67 signatures | > 100 (~3k gas per signature) |
+| Message Approval Batching | 1 | Configurable | > 500 (~34k gas per message) |
+| Storage Limit for Messages | Practically unlimited (> 2^64) | - | Practically unlimited |
+| Event Retention | 2 weeks | 2 months | Unlimited (tunable in node config) |
+
 ## Testing
 
-Unit tests for the gateway, and auth mechanism can be found [here](../../test/gateway/AxelarAmplifierGateway.js), and [here](../../test/governance/BaseWeightedMultisig.js), respectively, to use as reference for the implementation for another chain. Other than standard testing practices like unit tests, code coverage, Axelar Amplifier devnet e2e testing framework will support adding connectors for different chains. More details to come.
+Unit tests for the gateway, and auth mechanism can be found [here](../../test/gateway/AxelarAmplifierGateway.js), and [here](../../test/governance/BaseWeightedMultisig.js), respectively, to use as reference for testing the implementation for another chain. Other than standard testing practices like unit tests, code coverage, Axelar Amplifier devnet e2e testing framework will support adding connectors for different chains. More details to come.
```

Comment:
```suggestion
Unit tests for the gateway and auth mechanism can be found [here](../../test/gateway/AxelarAmplifierGateway.js), and [here](../../test/governance/BaseWeightedMultisig.js), respectively, to use as reference for testing the implementation for another chain. Other than standard testing practices like unit tests, code coverage, Axelar Amplifier devnet e2e testing framework will support adding connectors for different chains. More details to come.
```

Reply:
"Other than standard testing practices like unit tests, ... e2e testing framework will support adding connectors for different chains" - does this mean "**In addition** to standard testing practices..." ? Or does it mean that e2e testing will NOT include standard testing practices?

Reply:
in addition to, i'll switch to that